iob_soc.py

```diff
@@ -33,325 +34,325 @@
-                    split.instance("int_dbus_split_0"),
-                    split.instance("pbus_split_0"),
-                ],
+            (
+                {
+                    "corename": "UART0",
+                    "if_name": "rs232",
+                    "port": "rxd",
+                    "bits": [],
+                },
+                {
+                    "corename": "external",
+                    "if_name": "uart",
+                    "port": "uart_rxd_i",
+                    "bits": [],
+                },
             ),
-            iob_block_group(
-                name="mem",
-                description="Memory module",
-                blocks=[merge.instance("iob_merge_0"), merge.instance("iob_merge_1")],
+            (
+                {
+                    "corename": "UART0",
+                    "if_name": "rs232",
+                    "port": "cts",
+                    "bits": [],
+                },
+                {
+                    "corename": "external",
+                    "if_name": "uart",
+                    "port": "uart_cts_i",
+                    "bits": [],
+                },
             ),
-            iob_block_group(
-                name="peripheral",
-                description="Peripheral module",
-                blocks=self.peripherals,
+            (
+                {
+                    "corename": "UART0",
+                    "if_name": "rs232",
+                    "port": "rts",
+                    "bits": [],
+                },
+                {
+                    "corename": "external",
+                    "if_name": "uart",
+                    "port": "uart_rts_o",
+                    "bits": [],
+                },
             ),
         ]
+    )
+    # Number of external memory connections (will be filled automatically)
+    num_extmem_connections = -1
+    # This is a standard iob_module attribute, but needs to be defined after 'peripherals' because it depends on it
+    block_groups = [
+        iob_block_group(
+            name="cpu", description="CPU module", blocks=[cpu.instance("cpu_0")]
+        ),
+        iob_block_group(
+            name="bus_split",
+            description="Split modules for buses",
+            blocks=[
+                split.instance("ibus_split_0"),
+                split.instance("dbus_split_0"),
+                split.instance("int_dbus_split_0"),
+                split.instance("pbus_split_0"),
+            ],
+        ),
+        iob_block_group(
+            name="mem",
+            description="Memory module",
+            blocks=[merge.instance("iob_merge_0"), merge.instance("iob_merge_1")],
+        ),
+        iob_block_group(
+            name="peripheral",
+            description="Peripheral module",
+            blocks=self.peripherals,
+        ),
+    ]
 
-    def _setup(self, *args, is_top=True, **kwargs):
+    def __post_init__(self, *args, is_top=True, **kwargs):
```

Comment:
This is the new `__post_init__()` method (it was previously `_setup()`).
It is now responsible for setting up the the core right after initializing the attributes with the implicit `__init__()`.